packages/dds/tree/src/treeFactory.ts

```diff
@@ -51,17 +51,44 @@ export class TreeFactory implements IChannelFactory<ITree> {
 /**
  * SharedTree is a hierarchical data structure for collaboratively editing strongly typed JSON-like trees
  * of objects, arrays, and other data types.
- * @privateRemarks
- * Due to the dependency structure and the placement of that interface SharedObjectClass,
- * this interface implementation can not be recorded in the type here.
  * @public
  */
-export const SharedTree: ISharedObjectKind<ITree> = {
-	getFactory(): IChannelFactory<ITree> {
-		return new TreeFactory({});
-	},
+export const SharedTree: ISharedObjectKind<ITree> = configuredSharedTree({});
 
-	create(runtime: IFluidDataStoreRuntime, id?: string): ITree {
-		return runtime.createChannel(id, TreeFactory.type) as ITree;
-	},
-};
+/**
+ * {@link SharedTree} but allowing a non-default configuration.
+ * @remarks
+ * This is useful for debugging and testing to opt into extra validation or see if opting out of some optimizations fixes an issue.
+ * @example
+ * ```typescript
+ * import {
+ * 	ForestType,
+ * 	TreeCompressionStrategy,
+ * 	configuredSharedTree,
+ * 	typeboxValidator,
+ * 	// eslint-disable-next-line import/no-internal-modules
```

Comment:
Does this eslint directive need to be in the example?

Reply:
I considered removing it. I kinda like that it helps highlight that despite this pattern usually not being allowed, in this case (reaching into a package for on of our release tag rollups) is supported.

I don't have a strong opinion either way though, so if anyone has a preference, I'll go with that.

Reply:
No strong opinion either. Feel free to resolve without further changes.